src/aof.c

```diff
@@ -788,8 +811,17 @@ int openNewIncrAofForAppend(void) {
     /* Reset the aof_last_incr_size. */
     server.aof_last_incr_size = 0;
     /* Update `server.aof_manifest`. */
-    aofManifestFreeAndUpdate(temp_am);
+    if (temp_am) aofManifestFreeAndUpdate(temp_am);
     return C_OK;
+
+cleanup:
+    if (newfd != -1) close(newfd);
```

Comment:
let's to init `newfd` when declaring it. (make this code immune to future changes)